presto-common/src/main/java/com/facebook/presto/common/zorder/ZOrder.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common.zorder;
+
+import java.util.List;
+
+public interface ZOrder
+{
+    enum AddressType
+    {
+        INT, LONG
+    }
+
+    byte[] encodeToByteArray(List input);
+    long encodeToNumber(List input, AddressType type);
```

Comment:
This is interesting: if the return type is long already, we should give "type" operator. To make a clear class/interface structure, I will chat with you during the meeting today.